### DISCUSSION TITLE
1st of May, 2024 release

### DISCUSSION THREAD

**Discussion by tphakala** [tphakala/birdnet-go#137] [Announcements]
### 🚀 Features

 - Prometheus metrics support, endpoint reports BirdNET detections and application Golang runtime metrics - contributed by @aster1sk
 - Disk management by old audio capture cleanup - contributed by @isZumpo

### 🐛 Bug Fixes

- *(analysis)* File analysis restored
- *(capture)* Improve audio buffer write function time keeping
- *(datastore)* Refactor datastore Get, Delete and Save methods for efficient transaction and error handling
- *(datastore)* Refactor GetClipsQualifyingForRemoval method in interfaces.go for improved input validation and error handling
- *(birdweather)* Improve handling of HTTP Responses in UploadSoundscape to prevent possible panics
- *(birdweather)* Fixed PCM to WAV encoding for soundscape uploads
- *(birdweather)* Increase HTTP timeout to 45 seconds
- *(utils)* Do not report root user as missing from audio group
- *(tests)* Refactor createDatabase function in interfaces_test.go for improved error handling

### 💄 Enhancement

- *(audio)* Print selected audio capture device on realtime mode startup
- *(startup)* Enhance realtime mode startup message with system details to help troubleshooting

### 🚜 Refactor

- *(conf)* Remove unused Context struct from internal/conf/context.go
- *(processor)* Update range filter action to handle error when getting probable species

### 🏗️ Building

- *(deps)* Bump golang.org/x/crypto from 0.21.0 to 0.22.0
- *(deps)* Bump google.golang.org/protobuf from 1.32.0 to 1.33.0
- *(deps)* Bump golang.org/x/net from 0.21.0 to 0.23.0
- *(go)* Bump Go version from 1.21.6 to 1.22.2 in go.mod
- *(deps)* Bump labstack echo version from 4.11.4 to 4.12.0
- *(deps)* Bump gorm.io/gorm from 1.25.9 to 1.25.10
- *(deps)* Bump github.com/gen2brain/malgo from 0.11.21 to 0.11.22

### ⚙️ Miscellaneous Tasks

- Fix linter errors

### Github

- *(workflow)* Add tensorflow dependencies to golangci-lint

<hr /><em>This discussion was created from the release <a href='https://github.com/tphakala/birdnet-go/releases/tag/v0.5.2'>1st of May, 2024 release</a>.</em>

**Comment by ericleslie** [comment]
Awesome, thanks for the hard work. 

How does the file clean up work? Does it look at how full the partition is and then delete based on age? Just wondering if there's any benefit to getting a larger card so I can hold more recordings before they get deleted.

**Reply by tphakala** [reply]
At the moment retention policy is time based only, minEvictionHours sets time in hours to keep audio clips, any older clips than set here will be deleted. Cleanup is a bit limited for now.

I will add disk capacity based retention policy at some point (soon I hope). I will also add species specific settings so that clips of more common species get expunged first and some may be kept forever.
